modules/site-health/audit-autoloaded-options/load.php

```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * Module Name: Audit Autoloaded Options
+ * Description: Adds Autoloaded options checker in Site Health checks.
+ * Experimental: No
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Adds test to site health.
+ *
+ * @since 1.0.0
+ *
+ * @param array $tests Site Health Tests.
+ * @return array
+ */
+function perflab_aao_add_autoloaded_options_test( $tests ) {
+	$tests['direct']['autoloaded_options'] = array(
+		'label' => esc_html__( 'Autoloaded options', 'performance-lab' ),
+		'test'  => 'perflab_aao_autoloaded_options_test',
+	);
+	return $tests;
+}
+add_filter( 'site_status_tests', 'perflab_aao_add_autoloaded_options_test' );
+
+/**
+ * Callback for autoloaded_options test.
+ *
+ * @since 1.0.0
+ *
+ * @return array
+ */
+function perflab_aao_autoloaded_options_test() {
+
+	$autoloaded_options_size  = perflab_aao_autoloaded_options_size();
+	$autoloaded_options_count = count( wp_load_alloptions() );
+
+	$result = array(
+		'label'       => esc_html__( 'Autoloaded options', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => esc_html__( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+		/* translators: 1: Number of autoloaded options. 2.Autoloaded options size. */
+			'<p>' . esc_html__( 'The amount of %1$s autoloaded options (size: %2$s) in options table is acceptable.', 'performance-lab' ) . '</p>',
+			$autoloaded_options_count,
+			size_format( $autoloaded_options_size )
+		),
+		'actions'     => '',
+		'test'        => 'autoloaded_options',
+	);
+
+	/**
+	 * Hostings can modify its limits depending on their own requirements.
```

Comment:
See above, this would be better as "Filters the limits...". Also, which limits? The message right now does not mention autoloaded options in any way, so that's a bit too generic.

Reply:
updated. 

The message right now is the following:
> Your website uses %1$s autoloaded options (size: %2$s). Try to reduce the number of autoloaded options or performance will be affected.

We don't mention the default limit, sine in theory this feature would be used by different hosting providers that will set up their own limit.